File: rust/src/equipment/everything.rs
use crate::equipment::latex_table::{TableRow, ToTableRows};
use crate::equipment::{
    all_apparel, all_implements, all_magic_armor, all_magic_weapons, all_tools, latex_table,
};

pub fn everything_table() -> String {
    let mut rows: Vec<TableRow> = all_apparel(None)
        .iter()
        .map(|a| a.to_table_rows())
        .flatten()
        .collect();

    rows.append(
        &mut all_magic_armor(None)
            .iter()
            .map(|a| a.to_table_rows())
            .flatten()
            .collect(),
    );

    rows.append(
        &mut all_implements(None)
            .iter()
            .map(|a| a.to_table_rows())
            .flatten()
            .collect(),
    );

    rows.append(
        &mut all_magic_weapons(None)
            .iter()
            .map(|a| TableRow::from_item(a.item(), false, Some("Weapon".to_string())))
            .flatten()
            .collect(),
    );

    rows.append(
        &mut all_tools(Some(true), None)
            .iter()
            .map(|a| a.to_table_rows())
            .flatten()
            .collect(),
    );

    rows.append(
        &mut all_tools(Some(false), None)
            .iter()
            .map(|a| a.to_table_rows())
            .flatten()
            .collect(),
    );

    latex_table::standard_sort(&mut rows);
    let with_category = true;
    latex_table::longtable_percentile("All Items", rows, with_category)
}

File: typescript/src/equipment/latex/everything.ts
import { allMagicArmor } from '../data/magic_armor';
import { allMagicWeapons } from '../data/magic_weapons';
import { allImplements } from '../data/implements';
import { allApparel } from '../apparel';
import { allTools } from '../data/all_tools';
import * as table from './latex_table';
import { getToolCategoryName } from './tools';

export function generateEverythingTable(): string {
  const rows: table.TableRow[] = [];

  rows.push(
    ...allMagicArmor().flatMap((m) =>
      table.fromItem(m.item, false, m.kind === 'Body' ? 'Body armor' : 'Shield'),
    ),
  );
  rows.push(...allMagicWeapons().flatMap((m) => table.fromItem(m.item, false, 'Weapon')));
  rows.push(...allImplements().flatMap((m) => table.fromItem(m.item, false, m.kind)));
  rows.push(...allApparel(undefined).flatMap((a) => table.fromItem(a.item, false, a.kind)));
  rows.push(
    ...allTools().flatMap((t) =>
      table.fromItem(t.item, isConsumable(t), getToolCategoryName(t.category)),
    ),
  );

  table.standardSort(rows);
  return table.longtablePercentile({
    caption: 'All Items',
    rows,
    withCategory: true,
  });
}

function isConsumable(tool: any): boolean {
  if (typeof tool.category === 'string') {
    return (
      tool.category === 'Alchemical' || tool.category === 'Poison' || tool.category === 'Potion'
    );
  }
  return false;
}
